cloudformation/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,241 @@
+          - !Ref 'RAIDOptions'
+        - '2'
+    - !Condition 'UseRaidVol2'
+  UseRaidVol4: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '3'
+    - !Condition 'UseRaidVol3'
+  UseRaidVol5: !And
+    - !Not
+      - !Equals
+        - !Select
+          - '2'
+          - !Ref 'RAIDOptions'
+        - '4'
+    - !Condition 'UseRaidVol4'
+  # Duplicate of the ebs-substack code
+  UseEBSVol2: !Not
+    - !Equals
+      - !Ref 'NumberOfEBSVol'
+      - '1'
+  UseEBSVol3: !And
+    - !Not
+      - !Equals
+        - !Ref 'NumberOfEBSVol'
+        - '2'
+    - !Condition 'UseEBSVol2'
+  UseEBSVol4: !And
+    - !Not
+      - !Equals
+        - !Ref 'NumberOfEBSVol'
+        - '3'
+    - !Condition 'UseEBSVol3'
+  UseEBSVol5: !And
+    - !Not
+      - !Equals
+        - !Ref 'NumberOfEBSVol'
+        - '4'
+    - !Condition 'UseEBSVol4'
+Outputs: {}
+Parameters:
+  CreateFSXSubstack:
+    Description: Parameter telling if a FSx Substack has been created
+    Type: String
+  FSXFileSystemId:
+    Description: ID of the FSx volume used
+    Type: String
+  RAIDOptions:
+    Description: Comma Separated List of RAID related options, 8 parameters in total, [shared_dir,raid_type,num_of_raid_volumes,volume_type,volume_size,volume_iops,encrypted,ebs_kms_key_id]
+    Type: CommaDelimitedList
+  RAIDVolumesIDs:
+    Description: Volume IDs of the resulted RAID EBS volumes
+    Type: CommaDelimitedList
+  CreateRAIDSubstack:
+    Description: Parameter telling if a RAID Substack has been created
+    Type: String
+  EBSVolumesIDs:
+    Description: IDs of the EBS volumes used
+    Type: CommaDelimitedList
+  NumberOfEBSVol:
+    Description: Number of EBS Volumes the user requested, up to 5
+    Type: Number
+  EFSFileSystemID:
+    Description: ID of the EFS volume used
+    Type: String
+  ClusterName:
+    Description: Name of the cluster to which this dashboard belongs
+    Type: String
+  CreateEFSSubstack:
+    Description: Parameter telling if an EFS Substack has been created
+    Type: String
+  MasterInstanceID:
+    Description: ID of the Master instance
+    Type: AWS::EC2::Instance::Id
+Resources:
+  MasterCloudWatchDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+      DashboardName: !Join # Name like 'mycluster-Master'
+        - '-'
+        - - !Select
+            - '1'
+            - !Split
+              - 'parallelcluster-'
+              - !Ref 'ClusterName'
+          - 'Master'
+      DashboardBody: !Join
+        - ''
+        - - '{"widgets":['
+          - !Join # TODO: delete this join (since the one before is the same) if we do not change it back
+            - '' # Do not use comma join: Would make a problem when using If statements as Join(',', ['a','','b'])='a,,b'
+            # EC2 Instance metrics
+            - - '{"type":"text","x":0,"y":48,"width":24,"height":1,"properties":{"markdown":"\n#
+                      EC2 Instance Metrics\n"}}'
```

Comment:
"EC2 Instance Metrics" could be "Head Node Instance Metrics"

Reply:
Done